veniq/baselines/semi/recommend.py

```diff
@@ -0,0 +1,106 @@
+from typing import List, Tuple
+from tempfile import NamedTemporaryFile
+from functools import reduce
+from operator import itemgetter
+import os
+
+from veniq.utils.ast_builder import build_ast
+from veniq.ast_framework import AST, ASTNodeType
+from veniq.baselines.semi.rank_extraction_opportunities import \
+    rank_extraction_opportunities, ExtractionOpportunityGroup
+from veniq.baselines.semi.create_extraction_opportunities import \
+    create_extraction_opportunities
+from veniq.baselines.semi.extract_semantic import \
+    extract_method_statements_semantic
+from veniq.baselines.semi.filter_extraction_opportunities import \
+    filter_extraction_opportunities
+from veniq.baselines.semi._common_types import ExtractionOpportunity,\
+    OpportunityBenifit
+
+
+EMO = Tuple[int, int]
+
+
+def _add_class_decl_wrap(method_decl: List[str]) -> List[str]:
+    class_decl = ['class FakeClass {'] + method_decl + ['}']
+    return class_decl
+
+
+def _get_method_subtree(class_decl: List[str]) -> AST:
+    with NamedTemporaryFile(delete=False) as f:
+        _name = f.name
+        f.write('\n'.join(class_decl).encode())
+    ast = AST.build_from_javalang(build_ast(_name))
+    os.unlink(_name)
+    class_node = list(ast.get_proxy_nodes(ASTNodeType.CLASS_DECLARATION))[0]
```

Comment:
if there is no class declaration? enum, interface?

Reply:
i make sure there is a class declaration (in method recommend_for_method)